src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -30,31 +31,33 @@ private ConfigurationReader() {
     /**
      * Merge the given transaction log into the given configuration.
      *
-     * @param config         configuration to merge into
-     * @param reader         reader of the transaction log to read from
+     * @param config configuration to merge into
+     * @param reader reader of the transaction log to read from
      * @param forceTimestamp should ignore if the proposed timestamp is older than current
-     * @param mergeCondition Predicate that returns true if the provided Topic should be merged and false if not
+     * @param mergeCondition Predicate that returns true if the provided Topic should be merged and
+     *     false if not
      * @throws IOException if reading fails
      */
-    public static void mergeTLogInto(Configuration config, Reader reader, boolean forceTimestamp,
-                                     Predicate<Node> mergeCondition) throws IOException {
+    public static void mergeTLogInto(Configuration config, Reader reader, boolean forceTimestamp, Predicate<Node> mergeCondition)
+            throws IOException {
         mergeTLogInto(config, reader, forceTimestamp, mergeCondition, ConfigurationMode.WITH_VALUES);
     }
 
     /**
      * Merge the given transaction log into the given configuration.
      *
-     * @param config            configuration to merge into
-     * @param reader            reader of the transaction log to read from
-     * @param forceTimestamp    should ignore if the proposed timestamp is older than current
-     * @param mergeCondition    Predicate that returns true if the provided Topic should be merged and false if not
+     * @param config configuration to merge into
+     * @param reader reader of the transaction log to read from
+     * @param forceTimestamp should ignore if the proposed timestamp is older than current
+     * @param mergeCondition Predicate that returns true if the provided Topic should be merged and
+     *     false if not
      * @param configurationMode Configuration mode
      * @throws IOException if reading fails
      */
-    private static void mergeTLogInto(Configuration config, Reader reader, boolean forceTimestamp,
-                                      Predicate<Node> mergeCondition, ConfigurationMode configurationMode)
+    private static void mergeTLogInto(Configuration config, Reader reader, boolean forceTimestamp, Predicate<Node> mergeCondition, ConfigurationMode configurationMode)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.